### PR TITLE
Don't inherit from TerminalIPythonApp application class

### PR DESCRIPTION
**Pull request by takluyver**
The main practical effect of this is that jupyter console will look for its config in standard jupyter locations, rather than in IPython locations. But it's also conceptually better to disentangle jupyter_console from IPython.

Closes gh-31

### CODE REVIEW

**Comment by minrk** [comment]
That's a lot simpler than I thought it would be. +1.

**Comment by takluyver** [comment]
But not, unfortunately, quite as simple as I hoped. Debugging the failure...

**Comment by minrk** [comment]
You might want to just remove the `term_flags` bit

File: jupyter_console/app.py [hunk 1]
```diff
@@ -138,21 +138,27 @@ def handle_sigint(self, *args):
             # raise the KeyboardInterrupt if we aren't waiting for execution,
             # so that the interact loop advances, and prompt is redrawn, etc.
             raise KeyboardInterrupt
-    
-    def initialize(self, argv=None):
```
Comment (minrk) [inline jupyter_console/app.py]: You can't get rid of this guy, the NoStart catch is still needed.

File: jupyter_console/app.py [hunk 2]
```diff
@@ -7,6 +7,7 @@
 # Copyright (c) IPython Development Team.
 # Distributed under the terms of the Modified BSD License.
 
+import logging
 import signal
 
 from IPython.terminal.ipapp import TerminalIPythonApp, frontend_flags as term_flags
```
Comment (minrk) [inline jupyter_console/app.py]: Don't need these imports anymore

File: jupyter_console/app.py [hunk 3]
```diff
@@ -138,21 +138,27 @@ def handle_sigint(self, *args):
...
 
-    def init_code(self):
-        # no-op in the frontend, code gets run in the backend
-        pass
+    @catch_config_error
+    def initialize(self, argv=None):
+        """Do actions after construct, but before starting the app."""
+        super(ZMQTerminalIPythonApp, self).initialize(argv)
+        # create the shell
+        self.init_shell()
```
Comment (minrk) [inline jupyter_console/app.py]: You can add `if self._dispatching: return` after super(), and it should be happy, again.

File: jupyter_console/app.py (hunk 1 above)
Comment (minrk) [inline jupyter_console/app.py]: Strike that, there's a simpler way.

**Comment by minrk** [comment]
The cleaner subclass means the NoStart hack isn't needed anymore in init_shell. Something like this should do it: https://github.com/minrk/jupyter_console/commit/f96839c

**Comment by takluyver** [comment]
OK, processed your suggestions, and tests are passing again.

**Comment by minrk** [comment]
huzzah, thanks.

**Pull request merged**